iroh/src/node/builder.rs

```diff
@@ -309,7 +302,7 @@ where
     }
 
     /// Configure rpc endpoint, changing the type of the builder to the new endpoint type.
```

Comment:
This comment is not correct anymore, as the endpoint type doesn't change. Also I think we don't need to reconstruct the builder anymore here, it can just mutate itself, as the type stays

Reply:
We are using self and not &mut self everywhere in this builder, why change it here?

Reply:
no need to change the function signature: 

```rust
fn foo(mut self) -> Self {
  self.foo = true;
  self
}
```

Reply:
Ah yes, I ended up using `Self { self.foo=bla, ..self }`, old FP person that I am.